File: scalatest/src/org/game/Piece.scala
package org.game

/**
 * Created by IntelliJ IDEA.
 * User: knoxl
 * Date: Dec 5, 2010
 * Time: 5:19:22 PM
 * To change this template use File | Settings | File Templates.
 */
abstract class Piece(aName: String) {

  def name: String = aName

  private var mX, mY: Int = 0

  override def toString(): String = "(" + name + ")";

  private def validate(v : Int, message : String) = {
	require(v >= 0, message)
  }
  
  def setX(v: Int) = {
    validate(v, "x < 0")
    mX = v
  }

  def setY(v: Int) = {
    validate(v, "y < 0")
    mY = v
  }

  def x() = {
    mX;
  }

  def y() = {
    mY;
  }

  
}

File: scalatest/src/org/java/game/Piece.java
package org.java.game;


public class Piece {

	public static final Piece NONE = new Piece(".",0) {
	};
	
	private final String name;
	private final short id;
	
	public Piece(String name, int id) {
		this.name = name;
		this.id = (short)id;
	}

	public String toString() {
		return name;
	}

	public String getName() {
		return name;
	}


	public short getId() {
		return id;
	}
	
}
